package.json

```diff
@@ -29,10 +29,10 @@
     "prepack": "yarn build"
   },
   "dependencies": {
-    "@superfaceai/ast": "^1.0.0",
-    "@superfaceai/cli": "^1.0.1",
-    "@superfaceai/one-sdk": "^1.0.1",
-    "@superfaceai/parser": "^1.0.0",
+    "@superfaceai/ast": "^1.2.0",
+    "@superfaceai/one-sdk": "^1.4.1",
+    "@superfaceai/parser": "^1.2.0",
+    "@superfaceai/service-client": "^3.0.0",
```

Comment:
It would be safer to lock ast,parser and sdk version. Without it you can get newer version of ast/parser then sdk uses :/

Reply:
🤔 it is tricky here. I'll try to brainstorm here to turn my head around.

I see three ways how to specify version for ast, parser and OneSDK.

**1. Keep versions with caret**

Good thing about it is, it won't need updating OneService deps if bug fix for any of deps is released
But we may use newer version of parser and ast, then is used by OneSDK once usecase is performed

**2. OneSDK with caret, parser and ast specific version**

It is similar to 1. but here we risk opposite issues. If we don't update version of ast and parser, we may end with older version of parser and ast, than OneSDK uses while performing usecase.

**3. All specific version**

This seems as the safest option to avoid desyncing version of ast and parser. On other hand users who decide to use OneService as dependency in their project and just use OneService middleware, may have trouble to resolve versions if they use also OneSDK or any other dependency.

Also it means we will need to often update one-service dependencies to avoid locking users on older versions.

---

I am thinking if desynced versions can cause any problem here. And I can thing of only one scenario where it can cause issue. We use newer parser and ast, generate schema, which may have differences in handling values, so we generate schema with type X, but then it won't be accepted by perform.

Reply:
Yea, I wasn't sure about if there can be an issues with desynced versions. I am ok with using carets I just wanted to know how big problem it can cause:)